File: listings/probabilistic-graph.java
// Java
ClausesParser clausesParser = ClausesParser.withOperators(PROBLOG_OPERATORS);
Theory probabilisticTheory = clausesParser.parseTheory("...");
Solver problogSolver = Solver.problog.solverWithDefaultBuiltins(probabilisticTheory);
Struct goal = Struct.of("path", Var.of("From"), Var.of("To"));
var si = problogSolver.solve(goal, probabilistic(SolveOptions.allLazily())).iterator();
while (si.hasNext()) {
    Solution solution = si.next();
    if (solution.isYes()) {
        System.out.printf("yes: %s with probability %g\n", solution.getSolvedQuery(), getProbability(solution));
    }
}

File: listings/probabilistic-graph.kt
// Kotlin
val clausesParser = ClausesParser.withOperators(PROBLOG_OPERATORS)
val probabilisticTheory = clausesParser.parseTheory("...")
val problogSolver = Solver.problog.solverWithDefaultBuiltins(staticKb = probabilisticTheory)
val goal = Struct.of("path", Var.of("From"), Var.of("To"))
for (solution in problogSolver.solve(goal, SolveOptions.allLazily().probabilistic()))
    if (solution.isYes)
        println("yes: ${solution.solvedQuery} with probability ${solution.probability}")
